File: Angular_NET_CORE/ClientApp/src/app/products/models/product.ts
import { Category } from './category';
import { Size } from './size';

export interface Product{
    id: number;
    name: string;
    productCategory: Category;
    productSize: Size;
    quantity: number;
    price: number;
}

export interface ProductSelect extends Product{
    isSelected: boolean;
}

File: Angular_NET_CORE/Server/Common/DB_MODELS/Product.cs
using System;
using System.Collections.Generic;

namespace Common.DB_MODELS
{
    public class Product : BaseEntity
    {
        public int Quantity { set; get; }
        public double Price { set; get; }
        public string Name { set; get; }
        public DateTime CreatedAt { set; get; }
        public List<Order> Orders { set; get; } = new List<Order>();
        public int ProductSizeId { set; get; }
        public ProductSize ProductSize { set; get; }
        public int ProductCategoryId { set; get; }
        public ProductCategory ProductCategory { set; get; }
    }
}
